ArcGISRuntimeSDKQt_CppSamples/Routing/NavigateRoute/NavigateRoute.cpp

```diff
@@ -0,0 +1,271 @@
+#include "SimpleMarkerSymbol.h"
+#include "SimulatedLocationDataSource.h"
+#include "Stop.h"
+#include "TrackingDistance.h"
+#include "TrackingProgress.h"
+#include "TrackingStatus.h"
+#include "VoiceGuidance.h"
+
+#include <memory>
+#include <QList>
+#include <QGeoPositionInfo>
+#include <QTextToSpeech>
+#include <QTime>
+#include <QUrl>
+
+using namespace Esri::ArcGISRuntime;
+
+namespace  {
+const QUrl routeTaskUrl("https://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/Route");
+const Point conventionCenterPoint(-117.160386727, 32.706608, SpatialReference::wgs84());
+const Point memorialPoint(-117.173034, 32.712327, SpatialReference::wgs84());
+const Point aerospaceMuseumPoint(-117.147230, 32.730467, SpatialReference::wgs84());
+}
+
+NavigateRoute::NavigateRoute(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(Basemap::navigationVector(this), this))
+{
+  m_routeOverlay = new GraphicsOverlay(this);
+  m_routeTask = new RouteTask(routeTaskUrl, this);
+  m_speaker = new QTextToSpeech(this);
+}
+
+NavigateRoute::~NavigateRoute() = default;
+
+void NavigateRoute::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<NavigateRoute>("Esri.Samples", 1, 0, "NavigateRouteSample");
+}
+
+MapQuickView* NavigateRoute::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void NavigateRoute::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  m_mapView->graphicsOverlays()->append(m_routeOverlay);
+  connectRouteTaskSignals();
+
+  m_routeTask->load();
+
+  // add graphics for the predefined stops
+  SimpleMarkerSymbol* stopSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Diamond, Qt::red, 20, this);
+  m_routeOverlay->graphics()->append(new Graphic(conventionCenterPoint, stopSymbol, this));
+  m_routeOverlay->graphics()->append(new Graphic(memorialPoint, stopSymbol, this));
+  m_routeOverlay->graphics()->append(new Graphic(aerospaceMuseumPoint, stopSymbol, this));
+
+  emit mapViewChanged();
+}
+
+void NavigateRoute::connectRouteTaskSignals()
+{
+  connect(m_routeTask, &RouteTask::solveRouteCompleted, this, [this](QUuid, RouteResult routeResult)
+  {
+    if (routeResult.isEmpty())
+      return;
+    m_routeResult = routeResult;
+    m_route = m_routeResult.routes()[0];
+
+    m_mapView->setViewpointGeometry(m_route.routeGeometry(), 100);
+
+    // create a graphic to show the route
+    m_routeAheadGraphic = new Graphic(m_route.routeGeometry(), new SimpleLineSymbol(SimpleLineSymbolStyle::Dash, Qt::blue, 5, this), this);
+
+    // create a graphic to represent the route that's been traveled (initially empty).
+    m_routeTraveledGraphic = new Graphic(this);
+    m_routeTraveledGraphic->setSymbol(new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, Qt::cyan, 3, this));
```

Comment:
rather than doing this on a separate line you can pass the symbol directly to the graphic c'tor above